### PR TITLE
Build ref hvcf

### PR DESCRIPTION
**Pull request by lynnjo**
## Description

_Provide a summary of your changes including motivation and context.
If these changes fix a bug or resolves a feature request, be sure to link to that issue._

## Type of change

_What type of changes does your code introduce? Put an `x` in boxes that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [x] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have updated relevant documentation

## Changelog entry

_Please add a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
Added functionality to build a reference hvcf file
```

### CODE REVIEW

File: src/main/kotlin/net/maizegenetics/phgv2/cli/BuildRefVcf.kt [hunk 1]
```diff
@@ -1,13 +1,37 @@
...
+
+    var myRefSequence: Map<String, NucSeq>? = null
+
+    // refurl is not required.  If present, it will result in a ##reference header
+    // in the hvcf file.
+    val refurl by option(help = "URL where the reference FASTA file can be downloaded")
+        .default("")
+
     val bed by option(help = "BED file")
         .default("")
```
Comment (tcasstevens) [inline src/main/kotlin/net/maizegenetics/phgv2/cli/BuildRefVcf.kt:36]: I know you probably copied this from somewhere else. But I don't see the reason for have an empty default and then checking that it exists.  Can't this just be required()?  Same for all these parameters

File: src/main/kotlin/net/maizegenetics/phgv2/utils/VariantLoadingUtils.kt [hunk 2]
```diff
@@ -78,6 +78,8 @@ fun createGenericHeaderLineSet(): Set<VCFHeaderLine> {
     headerLines.add(VCFInfoHeaderLine("NS", 1, VCFHeaderLineType.Integer, "Number of Samples With Data"))
     headerLines.add(VCFInfoHeaderLine("AF", 3, VCFHeaderLineType.Integer, "Allele Frequency"))
     headerLines.add(VCFInfoHeaderLine("END", 1, VCFHeaderLineType.Integer, "Stop position of the interval"))
+    // These last 4 are needed for assembly g/hvcfs, but not for reference.  I am keeping them in as header
```
Comment (tcasstevens) [inline src/main/kotlin/net/maizegenetics/phgv2/utils/VariantLoadingUtils.kt:81]: For hvcfs? Or just gvcfs?

File: src/main/kotlin/net/maizegenetics/phgv2/cli/BuildRefVcf.kt (hunk 1 above)
Comment (lynnjo) [reply src/main/kotlin/net/maizegenetics/phgv2/cli/BuildRefVcf.kt:36]: @tcasstevens The refurl is not required.  I default it to "" so the value can be checked later.  If refurl != "", then a "##reference=..." header is added to the hvcf file.

Comment (zrm22) [reply src/main/kotlin/net/maizegenetics/phgv2/cli/BuildRefVcf.kt:36]: @tcasstevens I mentioned this in another PR, but I tried to leave off the required() portion but it was not working correctly without the default("").  I am not sure why this is, but it did matter.

File: src/main/kotlin/net/maizegenetics/phgv2/cli/BuildRefVcf.kt [hunk 3]
```diff
@@ -32,9 +64,139 @@ class BuildRefVcf : CliktCommand() {
...
+        myLogger.info("begin createRefHvcf,  refGenome=${refGenome}")
+
+
+        // Verify the bed file is good.
+        // If there are overlaps, throw an error and exit.
+        // Overlaps are printed to stdout.
+        val overlaps = verifyIntervalRanges(ranges)
+        if (overlaps.isNotEmpty()) {
+            // Overlaps not permitted.  User can fix via manually or via CreateValidIntervalsFilePlugin.  Throw error
+            overlaps.stream().forEach { entry: String -> myLogger.severe("BuildRefVcf:  range Overlap entry: $entry") }
```
Comment (zrm22) [inline src/main/kotlin/net/maizegenetics/phgv2/cli/BuildRefVcf.kt:85]: This is kotlin, no need for a .stream()

File: src/main/kotlin/net/maizegenetics/phgv2/cli/BuildRefVcf.kt [hunk 4]
```diff
@@ -32,9 +64,139 @@ class BuildRefVcf : CliktCommand() {
             }
         }
 
-
     override fun run() {
-//        TODO("Not yet implemented")
+
+        //createRefHvcf(bed,refFasta,refName,outputDir)
```
Comment (zrm22) [inline src/main/kotlin/net/maizegenetics/phgv2/cli/BuildRefVcf.kt:69]: Remove the commented out portion.

**Review by zrm22 — APPROVED** [review]

**Review by tcasstevens — APPROVED** [review]

**Pull request merged**